include/knowhere/feature.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+#ifndef FEATURE_H
+#define FEATURE_H
+
+// these features have been report to outside (milvus); pls sync the feature code when it needs to be changed.
+namespace knowhere::feature {
+// vector datatype support : binary
+constexpr uint64_t BINARY = 1UL << 0;
```

Comment:
can this be implemented as 
```c++
enum class FeatureFlag : uint64_t { 
    BINARY = ..., 
    FLOAT32 = ...,
    ... 
};
```
? Or is the point having a simple int value?

Reply:
The reason for choosing uint64 is that this field will be registered in Milvus's Go memory at startup to handle some control logic (pr : https://github.com/milvus-io/milvus/pull/36843). Therefore, a general data type was chosen instead of a complex structure.